build/mvn

```diff
@@ -116,7 +116,8 @@ install_zinc() {
 # the build/ folder
 install_scala() {
   # determine the Scala version used in Spark
-  local scala_version=`grep "scala.version" "${_DIR}/../pom.xml" | head -n1 | awk -F '[<>]' '{print $3}'`
+  local scala_binary_version=`grep "scala.binary.version" "${_DIR}/../pom.xml" | head -n1 | awk -F '[<>]' '{print $3}'`
+  local scala_version=`grep "scala.version" "${_DIR}/../pom.xml" | grep ${scala_binary_version} | head -n1 | awk -F '[<>]' '{print $3}'`
```

Comment:
Thank you for review, @viirya . 

Previously, it doesn't matched; `scala.binary.version=2.11` and `scala.version=2.12.7`.
Now, they will be matched; `scala.binary.version=2.11` and `scala.version=2.11.12`.
By default (without `change-scala-version.sh 2.11`), it's `scala.binary.version=2.12` and `scala.version=2.12.7`.

Reply:
Don't we need to update `dev/change-scala-version.sh` to let it update `<scala.version>`?

Reply:
I have above question because I think after current change, `scala.version` and `scala.binary.version` are still not matched in pom.xml, isn't

Or we just need to do as current change to download & install correct Scala and this is enough?

Reply:
@viirya .  I think you are confused about the meaning of `scala.version`.
In maven/sbt, `scala.version` is controlled by profile `-Pscala-2.11`. 
This `build/mvn` doesn't know the profile `-Pscala-2.11`. That's the problem this PR want to fix.

Reply:
oh, I see. Thanks @dongjoon-hyun